pandas/tests/base/test_ops.py

```diff
@@ -719,58 +719,63 @@ def test_drop_duplicates_series_vs_dataframe(self):
                 dropped_series = df[column].drop_duplicates(keep=keep)
                 tm.assert_frame_equal(dropped_frame, dropped_series.to_frame())
 
-    def test_fillna(self):
+    def test_fillna(self, index_or_series_obj):
         # # GH 11343
         # though Index.fillna and Series.fillna has separate impl,
         # test here to confirm these works as the same
 
-        for orig in self.objs:
-
-            o = orig.copy()
-            values = o.values
-
-            # values will not be changed
-            result = o.fillna(o.astype(object).values[0])
-            if isinstance(o, Index):
-                tm.assert_index_equal(o, result)
-            else:
-                tm.assert_series_equal(o, result)
-            # check shallow_copied
-            assert o is not result
+        obj = index_or_series_obj
+        if len(obj) == 0:
+            pytest.skip("Test doesn't make sense on empty data")
+        elif isinstance(obj, pd.MultiIndex):
```

Comment:
Can you use ```ABCMultiIndex``` from ```pandas.core.dtypes.generic```?

Reply:
Sure, done. There are more cases like that in the file, but I'd rather change these in a follow-up PR

Reply:
@SaturnFromTitan That would be great!